File: java/DataController.java
import java.util.Scanner;
import java.util.ArrayList;


public class DataController {

    private ArrayList<Shirt> container = new ArrayList<>();
    private Scanner sc = new Scanner(System.in);


    public DataController() {

    }

    public DataController(ArrayList<Shirt> src) {
        this.container = src;
    }

    public ArrayList<Shirt> getContainer() {
        return this.container;
    }

    public void setContainer(ArrayList<Shirt> container) {
        this.container = container;
    }

    public Shirt searchById(String id) {
        for(Shirt obj : this.container) {
            if(obj.getIdProduct().equals(id)) {
                return obj;
            }
        }
        return null;
    }

    public void createData() {
        String id, name, brand, size, material, gender, color, sleeveType;
        int price;

        System.out.println("\nPlease give attributes for new data:");
        System.out.print("Product ID : ");
        id = sc.next();
        sc.nextLine();
        System.out.print("Name        : ");
        name = sc.nextLine();
        System.out.print("Brand       : ");
        brand = sc.nextLine();
        System.out.print("Price       : ");
        price = sc.nextInt();
        sc.nextLine();
        System.out.print("Size        : ");
        size = sc.nextLine();
        System.out.print("Material    : ");
        material = sc.nextLine();
        System.out.print("Gender      : ");
        gender = sc.nextLine();
        System.out.print("Color       : ");
        color = sc.nextLine();
        System.out.print("Sleeve Type :");
        sleeveType = sc.nextLine();

        Shirt temp = new Shirt(id, name, brand, price, size, material, gender, color, sleeveType);
        this.container.add(temp);

        System.out.println("Data has been successfully created!");
        

    }

    public void modifyData() {
        String targetID;
        System.out.print("\nPlease select data to modify by ID : ");
        targetID = sc.next();

        Shirt target = this.searchById(targetID);
        if(target == null) {
            System.err.println("Data was not found!");
            return; 
        }

        System.out.println("\nData found. Please enter new data attributes");
        sc.nextLine();

        int price;
        String name, brand, size, material, gender, color, sleeveType;

        System.out.print("Name        : ");
        name = sc.nextLine();
        System.out.print("Brand       : ");
        brand = sc.nextLine();
        System.out.print("Price       : ");
        price = sc.nextInt();
        System.out.print("Size        : ");
        size = sc.nextLine();
        System.out.print("Material    : ");
        material = sc.nextLine();
        System.out.print("Gender      : ");
        gender = sc.nextLine();
        System.out.print("Color       : ");
        color = sc.nextLine();
        System.out.print("Sleeve Type :");
        sleeveType = sc.nextLine();

        target.setName(name);
        target.setBrand(brand);
        target.setPrice(price);
        target.setSize(size);
        target.setMaterial(material);
        target.setGender(gender);
        target.setColor(color);
        target.setSleeveType(sleeveType);

        System.out.println("Data successfully modified!");

    }

    public void deleteData() {
        String targetID;
        System.out.print("\nPlease select data to delete by ID : ");
        targetID = sc.next();

        Shirt target = this.searchById(targetID);
        if(target == null) {
            System.err.println("Data was not found!");
        }

        

        System.out.println("Data has been found.");
        System.out.println(target.getIdProduct() + ". " + target.getName());
        System.out.println("Confirm to delete? (y/n)");

        char del_confirm = sc.next().charAt(0);

        if(del_confirm == 'y') {
            this.container.remove(target);
        } else {
            System.out.println("Cancelled data deletion\n");
        }
        
        
    }
    
    public void showAll() {
        Table table = new Table();
        int numRows = this.container.size();
        
        String[][] data = new String[numRows + 1][9];
        
        data[0][0] = "Product ID";
        data[0][1] = "Name";
        data[0][2] = "Brand";
        data[0][3] = "Price";
        data[0][4] = "Size";
        data[0][5] = "Material";
        data[0][6] = "Gender";
        data[0][7] = "Color";
        data[0][8] = "Sleeve Type";





        for (int i = 0; i < numRows; i++) {
            Shirt Shirt = container.get(i);
            data[i + 1][0] = Shirt.getIdProduct();
            data[i + 1][1] = Shirt.getName();
            data[i + 1][2] = Shirt.getBrand();
            data[i + 1][3] = Integer.toString(Shirt.getPrice());
            data[i + 1][4] = Shirt.getSize();
            data[i + 1][5] = Shirt.getMaterial();
            data[i + 1][6] = Shirt.getGender();
            data[i + 1][7] = Shirt.getColor();
            data[i + 1][8] = Shirt.getSleeveType();



        }

        table.createRows(numRows + 1, 9, data);
        
    }

}

File: python/DataController.py
from Shirt import Shirt
from Table import Table

class DataController:
    
    __container = []

    def __init__(self):
        pass


    def getContainer(self):
        return self.__container
    
    def setContainer(self, container):
        self.__container = container

    def searchByID(self, id):
        found = False
        index = 0
        while not found and index < len(self.__container):
            if(self.__container[index].getId() == id):
                found = True
            else:
                index += 1
        
        if found:
            return self.__container[index]
        else:
            return None
        
    def createData(self):
        id = ""
        name = ""
        brand = ""
        price = 0
        size = ""
        material = ""
        gender = ""
        color = ""
        sleeve_type = ""

        print("Please give attributes for new data:")
        print("ID                 : ", end='')
        id = input()
        print("Name               : ", end='')
        name = input()
        print("Brand              : ", end='')
        brand = input()
        print("Price              : ", end='')
        price = int(input())
        print("Size               : ", end='')
        size = input()
        print("Material           : ", end='')
        material = input()
        print("Gender             : ", end='')
        gender = input()
        print("Color              : ", end='')
        color = input()
        print("Sleeve Type        : ", end='')
        sleeve_type = input()


        shirt = Shirt(id, name, brand, price, size, material, gender, color, sleeve_type)

        self.__container.append(shirt)

        print("Data has been successfully created!")

    
    def modifyData(self):
        targetID = ""
        print("PLease select data to modify by ID : ", end='')
        targetID = input()
        target = self.searchByID(targetID)

        if(target is None):
            print("Data was not found!")
            return
        
        print("Data found. Please enter new data attributes")

        name = ""
        brand = ""
        price = 0
        size = ""
        material = ""
        gender = ""
        color = ""
        sleeve_type = ""

        print("Please give attrubutes for new data:")
        print("Name               : ", end='')
        name = input()
        print("Brand              : ", end='')
        brand = input()
        print("Price              : ", end='')
        price = input()
        print("Size               : ", end='')
        size = input()
        print("Material           : ", end='')
        material = input()
        print("Gender             : ", end='')
        gender = input()
        print("Color              : ", end='')
        color = input()
        print("Sleeve Type        : ", end='')
        sleeve_type = input()


        target.setName(name)
        target.setBrand(brand)
        target.setPrice(price)
        target.setSize(size)
        target.setMaterial(material)
        target.setGender(gender)
        target.setColor(color)
        target.setSleeveType(sleeve_type)

        print("Data successfully modified!")

    def deleteData(self):
        
        targetID = ""
        print("Please select data to delete by ID : ", end='')
        targetID = input()
        target = self.searchByID(targetID)

        if(target is None):
            print("Data was not found!")
            return
        
        print("Data has been found.")
        print("{}. {}".format(target.getId(), target.getName()))
        print("Confirm to delete? (y/n)")
        del_confirm = input()
        if del_confirm == 'y':
            self.__container.remove(target)
            print("Data has successfully been deleted")
        else:
            print("Cancelled data deletion")
    

    def showAll(self):
        table = Table()
        data = [["" for j in range(9)] for i in range(len(self.__container) + 1) ] # + 1 untuk diawal diisi dengan header tabel

        data[0][0] = "Product ID"
        data[0][1] = "Name"
        data[0][2] = "Brand"
        data[0][3] = "Price"
        data[0][4] = "Size"
        data[0][5] = "Material"
        data[0][6] = "Gender"
        data[0][7] = "Color"
        data[0][8] = "Sleeve Type"

        for i, shirt in enumerate(self.__container):
            data[i + 1][0] = shirt.getId()
            data[i + 1][1] = shirt.getName()
            data[i + 1][2] = shirt.getBrand()
            data[i + 1][3] = str(shirt.getPrice())  
            data[i + 1][4] = shirt.getSize()
            data[i + 1][5] = shirt.getMaterial()
            data[i + 1][6] = shirt.getGender()
            data[i + 1][7] = shirt.getColor()
            data[i + 1][8] = shirt.getSleeveType()

        table.createRows(len(self.__container) + 1, 9, data)
        print("\n")
                
